### PR TITLE
changed all api calls to mongodb client for backup services

### PR DESCRIPTION
**Pull request by lovelgeorge99**
#1829 

## Summary by CodeRabbit

- **New Features**
	- Integrated support for MongoDB, PostgreSQL, and Redis databases.
	- Introduced a centralized database connection management system for MongoDB.

- **Refactor**
	- Streamlined donation management by replacing external API calls with direct MongoDB operations, enhancing performance and reliability.

- **Bug Fixes**
	- Improved error handling for database operations, ensuring accurate operation success checks.

### CODE REVIEW

File: src/adapters/donationSaveBackup/donationSaveBackupAdapter.ts [hunk 1]
```diff
@@ -1,68 +1,46 @@
...
-          'getNotImportedDonationsFromBackup error, status: ' + result.status,
-        );
-      }
-      return result.data.documents;
+      const result = await collection
+        .find({
+          // imported: { $exists: false },
+          // importError: { $exists: false },
+          projectId: 223479,
+        })
```
[the remark this answers is not in the document]
Comment (mohammadranjbarz) [reply src/adapters/donationSaveBackup/donationSaveBackupAdapter.ts]: @lovelgeorge99  please look at this comment

**Review by mohammadranjbarz — CHANGES_REQUESTED** [review]

File: src/adapters/donationSaveBackup/donationSaveBackupAdapter.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lovelgeorge99) [reply src/adapters/donationSaveBackup/donationSaveBackupAdapter.ts]: > @lovelgeorge99 please look at this comment
oh yes I forgot to uncomment the fields and remove the test projectId...i fixed in the new commit...thank you

**Review by mohammadranjbarz — APPROVED** [review]
Thanks @lovelgeorge99 LGTM

**Comment by CarlosQ96** [comment]
Hey guys is a env variable missing? @lovelgeorge99 @mohammadranjbarz or config? after this build the app breaks. error connecting to localhost port 6379, which made me think its redis. But now I think its related to this PR

**Pull request merged**